cadence/contracts/bridge/FlowEVMBridgeCustomAssociationTypes.cdc

```diff
@@ -0,0 +1,149 @@
+import "NonFungibleToken"
+import "CrossVMMetadataViews"
+import "EVM"
+
+/// This contract defines types required for custom cross-VM associations as used in FlowEVMBridgeCustomAssociation
+/// and in EVM-native cross-VM NFTs.
+///
+access(all) contract FlowEVMBridgeCustomAssociationTypes {
+
+    access(all) entitlement FulfillFromEVM
+
+    /// Resource interface used by EVM-native NFT collections allowing for the fulfillment of NFTs from EVM into Cadence
+    ///
+    access(all) resource interface NFTFulfillmentMinter {
+        /// Getter for the type of NFT that's fulfilled by this implementation
+        ///
+        access(all) view fun getFulfilledType(): Type
+
+        /// Called by the VM bridge when moving NFTs from EVM into Cadence if the NFT is not in escrow. Since such NFTs
+        /// are EVM-native, they are distributed in EVM. On the Cadence side, those NFTs are handled by a mint & escrow
+        /// pattern. On moving to EVM, the NFTs are minted if not in escrow at the time of bridging.
+        ///
+        /// @param id: The id of the token being fulfilled from EVM
+        ///
+        /// @return The NFT fulfilled from EVM as its Cadence implementation
+        ///
+        access(FulfillFromEVM)
+        fun fulfillFromEVM(id: UInt256): @{NonFungibleToken.NFT} {
```

Comment:
Could use feedback on the assumptions about id in the pre & post conditions

Reply:
What assumptions do you mean? Do we require that all EVM NFT ids be less than UInt64 max? What you have seems fine to me